src/main/java/com/dabsquared/gitlabjenkins/util/CommitStatusUpdater.java

```diff
@@ -64,7 +72,15 @@ private static void printf(TaskListener listener, String message, Object... args
     }
 
     private static String getBuildRevision(Run<?, ?> build) {
-        return build.getAction(BuildData.class).getLastBuiltRevision().getSha1String();
+        BuildData action = build.getAction(BuildData.class);
+        try {
+            Field lastBuildField = action.getClass().getDeclaredField("lastBuild");
+            lastBuildField.setAccessible(true);
+            Build lastBuild = (Build) lastBuildField.get(action);
+            return lastBuild.getMarked().getSha1String();
+        } catch (NoSuchFieldException | IllegalAccessException e) {
+            return null;
+        }
     }
```

Comment:
You could try the method getLastBuild(ObjectId sha1) of the BuildData and pass the result of getLastBuildRevision().getSha1(). If this works we would get rid of the ugly reflection stuff.